### PR TITLE
Tweak member page filtering

### LINKED ISSUE #1583: Members Page Regression
The members page all members tab no longer shows all the members in the colony, but now erroneously shows just the members that also have reputation.

This is not correct, as that should be shown only by the "Contributors" tab, while "All Members"  should show all the members of a specific colony

Screenshot from 2023-12-19 13-44-55
Screenshot from 2023-12-19 13-45-03

This is not a database issue, as the data is available in the members query _(`MembersContext`)_

Screenshot from 2023-12-19 13-53-50

But some hook or util is filtering out members more that it should

Screenshot from 2023-12-19 13-54-22

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
Fixed members without rep or roles disappearing from the `All members` tab. Now they should appear when you are not filtering by a team. The rest of the filters should work as expected (Tested various combinations, let me know if you find any that is not working as expected and I'll fix it in this PR)

Resolves #1583

### CODE REVIEW

**Comment by rdig** [comment]
This needs rebasing once again, since after https://github.com/JoinColony/colonyCDapp/pull/1540 merged it can no longer be tested

**Comment by ArmandoGraterol** [comment]
It is done Mr. @rdig

**Review by bassgeta — DISMISSED** [review]

File: src/hooks/members/useMemberFilters.ts [hunk 1]
```diff
@@ -34,14 +34,56 @@ const useMemberFilters = ({
...
+          : nativeDomainIds.map((id) => `${colonyAddress}_${id}`),
+      ),
+    [nativeDomainIds, colonyAddress, selectedTeam],
+  );
+
+  // Always include the root domain, since if the user has a permission in root, they have it in all domains
+  const permissionsDomainIds = useMemo(
+    () => new Set([`${colonyAddress}_${Id.RootDomain}`, ...databaseDomainIds]),
+    [databaseDomainIds, colonyAddress],
+  );
```
Comment (jakubcolony) [inline src/hooks/members/useMemberFilters.ts:51]: Could we use `getDomainDatabaseId` util here?

Comment (jakubcolony) [reply src/hooks/members/useMemberFilters.ts:51]: And is the useMemo necessary?

**Review by jakubcolony — APPROVED** [review]
Nice work @ArmandoGraterol!

File: src/hooks/members/useMemberFilters.ts (hunk 1 above)
Comment (ArmandoGraterol) [reply src/hooks/members/useMemberFilters.ts]: I think so, these const are being used by other parts of the code that are memorized so if I remove the `useMemo` here it tells me that it needs it to not re-render the other functions.

Comment (jakubcolony) [reply src/hooks/members/useMemberFilters.ts]: Fair enough if that's the case, although I bet wherever the "useMemo chain" starts it's not needed there either

**Review by jakubcolony — APPROVED** [review]

**Review by rdig — APPROVED** [review]
Seems to be working good. Nice work Armando

**Review by melyndav — APPROVED** [review]
@ArmandoGraterol thanks for fixing this bug! Works as expected. :)

**Pull request merged**